x-pack/plugins/case/server/services/index.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  Logger,
+  SavedObject,
+  SavedObjectsClientContract,
+  SavedObjectsFindResponse,
+  SavedObjectsUpdateResponse,
+  SavedObjectReference,
+} from 'kibana/server';
+import { CASE_COMMENT_SAVED_OBJECT, CASE_SAVED_OBJECT } from '../constants';
+import {
+  NewCaseFormatted,
+  NewCommentFormatted,
+  UpdatedCaseType,
+  UpdatedCommentType,
+} from '../routes/api/types';
+import { PluginSetupContract as SecurityPluginSetup } from '../../../security/server';
+import { AuthenticatedUser } from '../../../security/common/model';
+
+interface ClientArgs {
+  client: SavedObjectsClientContract;
+}
+
+interface GetCaseArgs extends ClientArgs {
+  caseId: string;
+}
+interface GetCommentArgs extends ClientArgs {
+  commentId: string;
+}
+interface PostCaseArgs extends ClientArgs {
+  attributes: NewCaseFormatted;
+}
+
+interface PostCommentArgs extends ClientArgs {
+  attributes: NewCommentFormatted;
+  references: SavedObjectReference[];
+}
+interface UpdateCaseArgs extends ClientArgs {
+  caseId: string;
+  updatedAttributes: UpdatedCaseType;
+}
+interface UpdateCommentArgs extends ClientArgs {
+  commentId: string;
+  updatedAttributes: UpdatedCommentType;
+}
+
+interface GetUserArgs {
+  request: KibanaRequest;
+  response: KibanaResponseFactory;
+}
+
+interface CaseServiceDeps {
+  authentication: SecurityPluginSetup['authc'] | null;
```

Comment:
Same comment as before, since `security` is marked as a required plugin, you should always have an instance of the `authentication` service here.

If security is an optional dependency, then we should update `kibana.json` and your `PluginsSetup` interface to mark it as such.